### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by nacharya1**
### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -111,6 +111,14 @@ The scripts below are used to build, test, and lint within the project.
...
+#### Plugins
+We appreciate your interest in extending Semantic Kernel's functionality through plugins. However, we want to clarify our approach to hosting plugins within our GitHub repository. To maintain a clean and manageable codebase, we will not be hosting plugins directly in the Semantic Kernel GitHub repository.
+Instead, we encourage contributors to host their plugin code in separate repositories under their own GitHub accounts or organization. You can then provide a link to your plugin repository in the relevant discussions, issues, or documentation within the Semantic Kernel repository. This approach ensures that each plugin can be maintained independently and allows for easier tracking of updates and issues specific to each plugin.
+#### Memory Connectors
+For memory connectors, while we won't be directly adding hosting for them within the Semantic Kernel repository, we highly recommend building memory connectors as separate plugins. Memory connectors play a crucial role in interfacing with external memory systems, and treating them as plugins enhances modularity and maintainability.
+
```
Comment (dluc) [inline CONTRIBUTING.md:121]:
```suggestion
### Adding Plugins and Memory Connectors

When considering contributions to plugins and memory connectors for Semantic
Kernel, please note the following guidelines:

#### Plugins

We appreciate your interest in extending Semantic Kernel's functionality through
plugins. However, we want to clarify our approach to hosting plugins within our
GitHub repository. To maintain a clean and manageable codebase, we will not be
hosting plugins directly in the Semantic Kernel GitHub repository.
Instead, we encourage contributors to host their plugin code in separate
repositories under their own GitHub accounts or organization. You can then
provide a link to your plugin repository in the relevant discussions, issues,
or documentation within the Semantic Kernel repository. This approach ensures
that each plugin can be maintained independently and allows for easier tracking
of updates and issues specific to each plugin.

#### Memory Connectors

For memory connectors, while we won't be directly adding hosting for them within
the Semantic Kernel repository, we highly recommend building memory connectors
as separate plugins. Memory connectors play a crucial role in interfacing with
external memory systems, and treating them as plugins enhances modularity and
maintainability.

```

**Review by dluc — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -111,6 +111,32 @@ The scripts below are used to build, test, and lint within the project.
...
+We appreciate your interest in extending Semantic Kernel's functionality through
+plugins. However, we want to clarify our approach to hosting plugins within our
+GitHub repository. To maintain a clean and manageable codebase, we will not be
+hosting plugins directly in the Semantic Kernel GitHub repository.
+Instead, we encourage contributors to host their plugin code in separate
+repositories under their own GitHub accounts or organization. You can then
+provide a link to your plugin repository in the relevant discussions, issues,
+or documentation within the Semantic Kernel repository. This approach ensures
+that each plugin can be maintained independently and allows for easier tracking
+of updates and issues specific to each plugin.
```
Comment (dluc) [inline CONTRIBUTING.md:130]: nit: a lot of "repository" repetitions

**Review by dluc — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (dluc) [reply CONTRIBUTING.md:130]: image

**Review by lemillermicrosoft — APPROVED** [review]

**Comment by nacharya1** [comment]
@shawncal any thoughts/feedback on the wording/approach

**Review by RogerBarreto — APPROVED** [review]

**Review by awharrison-28 — APPROVED** [review]

**Pull request merged**